src/components/recommendation.tsx

```diff
@@ -0,0 +1,75 @@
+import { useQuery } from "@tanstack/react-query";
+
+import { mockFetch } from "@/lib/mock-fetch";
+import type { RecommendationResponse } from "@/types/api-types";
+import type { Chat } from "@/types/chat";
+
+import { PromochatorIcon } from "./chat";
+import { Supervisor } from "./supervisor";
+
+function useRecommendationQuery(
+  chat: Chat,
+  updateChat: (uuid: string, _chat: Partial<Chat>) => void,
+) {
+  return useQuery({
+    queryKey: ["recommendation", chat.uuid],
+    queryFn: async () => {
+      const response = await mockFetch("/api/recommend");
+      // const response = await fetch("/api/recommend", {
+      //   method: "POST",
+      //   body: JSON.stringify({
+      //     input: { question: chat.prompt, faculty: "" },
+      //   }),
+      // });
+
+      const data = (await response.json()) as RecommendationResponse;
+      updateChat(chat.uuid, { recommendation: data.recommendation });
+      return data;
+    },
+    enabled: !chat.recommendation,
+    retry: false,
+  });
+}
+
+export function Recommendation({
+  chat,
+  updateChat,
+}: {
+  chat: Chat;
+  updateChat: (uuid: string, _chat: Partial<Chat>) => void;
+}) {
+  const { isLoading, error } = useRecommendationQuery(chat, updateChat);
+
+  if (isLoading) {
+    //TODO Loading UI
+    return <div>Loading...</div>;
+  }
+
+  if (error) {
+    //TODO Error
+    return <div>Error loading recommendations</div>;
+  }
+
+  return (
+    <div className="w-full max-w-7xl space-y-4 p-8">
+      <div className="flex items-center gap-4">
+        <PromochatorIcon
+          imageWidth={40}
+          imageHeight={40}
+          imageClassName="py-2 px-1"
+        />
+        <p className="rounded-2xl bg-message-primary px-4 py-3">
+          {chat.recommendation?.hello_message}
+        </p>
+      </div>
+      {chat.recommendation?.recommended_supervisors.map((supervisor) => (
+        <Supervisor
+          key={supervisor.name}
+          name={supervisor.name}
+          faculty={supervisor.faculty}
+          papers={supervisor.papers}
+        />
+      ))}
+    </div>
```

Comment:
**issue:** Also yet another problem i missed in the supervisor PR 😭 - `supervisor.tsx` returns an `<Accordion>` with a single `<AccordionItem>` which shows data from the props in `<AccordionContent>`. This is an incorrect implementation. In shadcn, we wrap **ALL** accordion items in **ONE SINGLE** `<Accordion>` component. Fixing this requires the suggested change below and changing `supervisor.tsx` so that it only returns `<AccordionItem>` contents
https://ui.shadcn.com/docs/components/accordion
```suggestion
      <Accordion>
        {chat.recommendation?.recommended_supervisors.map((supervisor) => (
          // this returns just <AccordionItem>... rest of the code<AccordionItem/>
          <Supervisor
            key={supervisor.name}
            name={supervisor.name}
            faculty={supervisor.faculty}
            papers={supervisor.papers}
          />
        ))}
      </Accordion>
```
@mejsiejdev

Reply:
yup, also noticed that but decided not to change it as it works xd
If @mejsiejdev has time then he can change it in separate PR